### PR TITLE
Added getBridge() to DiscoveryService 

### PR DESCRIPTION
**Pull request by afuechsel**
Added getBridge() to DiscoveryService to indicate the bridge UID that actually created this discovery service.

Before this PR one could approve a hue bridge from the inbox which in turn added all hue lights of this bridge to the inbox. When pairing another hue bridge and removing it later all hue lights have been removed from the inbox, regardless of whether they are from the first or the second bridge. With this PR the results from the first and still existing bridge are no longer removed if another bridge is removed. 

Signed-off-by: Andre Fuechsel <andre.fuechsel@telekom.de>

### CODE REVIEW

File: bundles/config/org.eclipse.smarthome.config.discovery.test/src/test/groovy/org/eclipse/smarthome/config/setup/test/discovery/DiscoveryServiceMockOfBridge.groovy [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.eclipse.smarthome.config.setup.test.discovery
```
Comment (sjka) [inline bundles/config/org.eclipse.smarthome.config.discovery.test/src/test/groovy/org/eclipse/smarthome/config/setup/test/discovery/DiscoveryServiceMockOfBridge.groovy]: Oh noooo, pleeease, no NEW groovy tests!

File: bundles/config/org.eclipse.smarthome.config.discovery.test/src/test/groovy/org/eclipse/smarthome/config/setup/test/discovery/DiscoveryServiceRegistryOSGITest.groovy [hunk 2]
```diff
@@ -323,6 +339,55 @@ class DiscoveryServiceRegistryOSGITest extends OSGiTest {
         anotherDiscoveryServiceMockForBinding1.abortScan()
     }
 
+    @Test
+    void 'assert that removeOlderResults removes only the entries of the specified bridge' () {
```
Comment (sjka) [inline bundles/config/org.eclipse.smarthome.config.discovery.test/src/test/groovy/org/eclipse/smarthome/config/setup/test/discovery/DiscoveryServiceRegistryOSGITest.groovy]: Groovy. See above.

File: bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryService.java [hunk 3]
```diff
@@ -50,22 +51,32 @@
...
-    public Collection<ThingTypeUID> getSupportedThingTypes();
+    Collection<ThingTypeUID> getSupportedThingTypes();
+
+    /**
+     * If the discovery service discovers things for a specific bridge only, this method
+     * would return the UID of the bridge.
+     *
+     * @return UID of the bridge that has started this discovery service, may be {@null}
+     */
+    default @Nullable ThingUID getBridge() {
```
Comment (sjka) [inline bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryService.java]: What if a discovery service _can_ discover things for several bridge instances - and you still would like to get rid of results for a single bridge?

File: bundles/config/org.eclipse.smarthome.config.discovery.test/src/test/groovy/org/eclipse/smarthome/config/setup/test/discovery/DiscoveryServiceMockOfBridge.groovy (hunk 1 above)
Comment (afuechsel) [reply bundles/config/org.eclipse.smarthome.config.discovery.test/src/test/groovy/org/eclipse/smarthome/config/setup/test/discovery/DiscoveryServiceMockOfBridge.groovy]: It's just a copy of an existing test modified to check the new behavior. As the whole infrastructure for the DiscoveryService tests is in Groovy I see no benefit in duplicating it to Java.

File: bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryService.java (hunk 3 above)
Comment (afuechsel) [reply bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryService.java]: Isn't it a problem that a discovery service is able to discover things for more than one bridge instance? I mean, having a bridge that creates a discovery service, I only want results of this service for this bridge

**Comment by sjka** [comment]
Why don't you simply extends the `DiscoveryListener` interface with a method signature that also accepts a bridge's UID?

File: bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryService.java (hunk 3 above)
Comment (sjka) [reply bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryService.java]: https://github.com/openhab/openhab2-addons/blob/e2befee912df9c1f65f9423a166d96a62ac8e7cc/addons/binding/org.openhab.binding.folding/src/main/java/org/openhab/binding/folding/internal/discovery/FoldingSlotDiscoveryService.java

Comment (sjka) [reply bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryService.java]: https://github.com/openhab/openhab2-addons/blob/739a8bb2126c22d36830b21908b051ffe3399d1e/addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/discovery/ZoneDiscoveryService.java

**Comment by cdjackson** [comment]
> Isn't it a problem that a discovery service is able to discover things for more than one bridge instance? I mean, having a bridge that creates a discovery service, I only want results of this service for this bridge

Maybe not directly linked, but since you talk about multiple bridges and only getting discovery results from one service, I'll just link this other issue...

Discovery should be able to activate per bridge, not per binding #3605

**Comment by triller-telekom** [comment]
LGTM, except for the new groovy tests :(

**Comment by afuechsel** [comment]
@triller-telekom well, as I said above I just copied an existing test and tailored it to suit my needs. Would it really be a good idea, to re-create the whole test infrastructure of these OSGi tests just to add another, nearly similar test? Or should I convert everything into Java? That's far beyond the scope of this PR. I know, that ESH is going away from Groovy back to Java although this makes things more difficult IMHO and I would create plain new tests in Java as required, but not when I can reuse existing test infrastructure.

**Comment by sjka** [comment]
FTR: I created #4704 where I ported the complete test to Java for you.

**Comment by kaikreuzer** [comment]
FTR: #4704 has been merged and this PR here needs to be rebased and adapted.

**Comment by afuechsel** [comment]
Rebased, ported my own test to Java

**Comment by kaikreuzer** [comment]
I'll restart Travis.

**Comment by kaikreuzer** [comment]
This time it's the DMX tests :-(

**Comment by afuechsel** [comment]
@kaikreuzer how shall we proceed here? I am unable to reproduce these test failures.

**Comment by sjka** [comment]
PS: The IP validation seems to fail because of the merge commit which will get squashed away anyway.

**Pull request merged**